src/UI/Implementation/Component/Button/Renderer.php

```diff
@@ -79,6 +90,38 @@ protected function maybeRenderId(Component\Component $component, $tpl) {
 		}
 	}
 
+	protected function renderMonth(Component\Button\Month $component, RendererInterface $default_renderer) {
+		$def = $component->getDefault();
+
+		for ($i = 1; $i<=12; $i++)
+		{
+			$this->toJS(array("month_".str_pad($i, 2, "0", STR_PAD_LEFT)."_long"));
+		}
+
+		$tpl = $this->getTemplate("tpl.month.html", true, true);
+
+		$month = explode("-", $def);
+		$tpl->setVariable("DEFAULT_LABEL", $this->txt("month_".str_pad($month[0], 2, "0", STR_PAD_LEFT)."_long")." ".$month[1]);
+
+		include_once("./Services/Calendar/classes/class.ilCalendarUtil.php");
+		\ilCalendarUtil::initDateTimePicker();
```

Comment:
And keep the same lines in Services/Calendar? I can live with this redundancy in this case, too. But in general we need something to avoid these cases I think. Especially, if more js/css files are involved this may become messy. For PHP libs the autoloader takes care of avoiding these long pathes in require/includes. For Javascript there is not autoloading...